default.nix

```diff
@@ -40,7 +40,7 @@ let
     overlays = [ overlay ];
   };
 in {
-  inherit (pkgs) cabal-install grpc mosquitto;
+  inherit (pkgs) cabal-install grpc haskell-language-server mosquitto;
   inherit (pkgs.haskell.packages."${ghc}") 
     grpc-mqtt hp2pretty proto3-suite threadscope;
```

Comment:
Why do we need all of these extra packages and tools bundled in this derivation output? Isn't having them in the `shell.nix` enough?

Reply:
I'm not sure how I would provide the packages to `shell.nix` without exposing them in `default.nix`, what do you recommend?

Reply:
Oh I see you moved the `nixpkg` stuff into `default.nix`. If you pulled your definition of `pkgs` into a separate file, then you could just import that into both `default.nix` and `shell.nix`.

Something like this:

`default.nix`
```nix
let 
  pkgs = import nix/pkgs.nix {};
in
  pkgs.haskell.packages."${ghc}".grpc-mqtt
```

`shell.nix`
```nix
let 
  pkgs = import nix/pkgs.nix {};

in pkgs.grpc-mqtt.env.overrideAttrs (old: {
  buildInputs = (old.buildAttrs or []) ++ [
    pkgs.cabal-install
    pkgs.grpc
    pkgs.haskell-language-server
    pkgs.hp2pretty
    pkgs.mosquitto
    pkgs.proto3-suite
    pkgs.threadscope
  ];
})
```

Reply:
In this case, would it not be better just to change `release.nix` to be
```nix
let 
  pkgs = import nix/pkgs.nix {};
in
  pkgs.haskell.packages."${ghc}".grpc-mqtt
  ```